### DISCUSSION TITLE
Compiling error with Debian Trixi, ccmake configure says "Unable to find iwlib.h"

### DISCUSSION THREAD

**Discussion by Pitido** [brndnmtthws/conky#1796] [Q&A]
Hello,

I use Debian Trixi and like to compile conky by using github/git.
Works fine for the first step with a standard build.

Now I like to add WLAN and switch Build_Wlan to "ON".

CCMAKE tells me:

`CMake Error at cmake/ConkyPlatformChecks.cmake:293 (message):
   Unable to find iwlib.h
 Call Stack (most recent call first):
   CMakeLists.txt:35 (include)
 Configuring incomplete, errors occurred!`

But mlocate tells:

`locate iwlib.h 
/usr/include/iwlib.h
`
`apt-file search iwlib.h
libghc-iwlib-doc: /usr/lib/ghc-doc/haddock/iwlib-0.1.2/iwlib.haddock
libiw-dev: /usr/include/iwlib.h
`

`sudo dpkg -l | grep libiw-dev
ii  libiw-dev:amd64                                30~pre9-16+b1                     amd64        Wireless tools - development files
`
I would be glad for any hint!

KR

Pitido

**Comment by Caellian** [comment]
That's weird. When you run cmake, does `INCLUDE_DIRECTORIES` include `/usr/include`? I mean, it's a pretty standard path on linux so it _should_ be there.

As a workaround, you can set the paths for `IWLIB_H` (iwlib.h) and `IWLIB_LIB` (libiw.so) manually and remove the checks:

So in `cmake/ConkyPlatformChecks.cmake`:
```
check_include_files(iwlib.h IWLIB_H)

if(NOT IWLIB_H)
  message(FATAL_ERROR "Unable to find iwlib.h")
endif(NOT IWLIB_H)

find_library(IWLIB_LIB NAMES iw)

if(NOT IWLIB_LIB)
  message(FATAL_ERROR "Unable to find libiw.so")
endif(NOT IWLIB_LIB)
```
becomes:
```
set(IWLIB_H /usr/include/iwlib.h)
set(IWLIB_LIB /usr/lib/libiw.so) # or other path if it's different

set(conky_libs ${conky_libs} ${IWLIB_LIB})
check_function_exists(iw_sockets_open IWLIB_SOCKETS_OPEN_FUNC)
```

**Reply by Pitido** [reply]
I run cmake with the gui ccmake.
In ccmake I found no `INCLUDE_DIRECTORIES`, but several other entries, which are maintained with /user/include.

**Reply by Pitido** [reply]
Now `configure` and `generate` works after your proposed workaround.
But `make` gives an error (sorry, here in German)

```
make[2]: *** Keine Regel vorhanden, um das Ziel „/usr/lib/libiw.so“, 
  benötigt von „src/conky“, zu erstellen.  Schluss.
make[2]: *** Es wird auf noch nicht beendete Prozesse gewartet....
[ 97%] Building CXX object src/CMakeFiles/conky.dir/nc.cc.o
make[1]: *** [CMakeFiles/Makefile2:307: src/CMakeFiles/conky.dir/all] Fehler 2
make: *** [Makefile:156: all] Fehler 2

```

**Reply by Caellian** [reply]
Just to make sure, `/usr/lib/libiw.so` exists? What's the output of `apt-file search libiw.so`?

**Comment by Pitido** [comment]
Hi,

thank you for you question.

'apt-file search libiw.so
libiw-dev: /usr/lib/x86_64-linux-gnu/libiw.so
libiw30: /usr/lib/x86_64-linux-gnu/libiw.so.30'


'$ locate libiw.so
/usr/lib/x86_64-linux-gnu/libiw.so
/usr/lib/x86_64-linux-gnu/libiw.so.30'

After 'ln -s x86_64-linux-gnu/libiw.so /usr/lib/libiw.so' it works!

But I like to avoid to change the system, can I change it in cmake?

Again, thank you so much for your help!

Kind regards

Pitido

**Reply by Caellian** [reply]
Yes, use `set(IWLIB_LIB /usr/lib/x86_64-linux-gnu/libiw.so)` instead of `set(IWLIB_LIB /usr/lib/libiw.so)`.
"ln -s" shouldn't be needed.

**Reply by brndnmtthws** [reply]
You can also try setting [`CMAKE_PREFIX_PATH`](https://cmake.org/cmake/help/latest/variable/CMAKE_PREFIX_PATH.html#variable:CMAKE_PREFIX_PATH) to `/usr/lib/x86_64-linux-gnu`, though I'm unsure if the `check_include_files()` will correctly handle that. This issue is a odd as the debian package doesn't appear to do anything particularly special to make this work correctly with WLAN enabled (see https://packages.debian.org/sid/conky-all).

**Reply by Pitido** [reply]
Ok, thank you all!
